### ISSUE TITLE
Make shell agnostic (ie, support bash, sh, zsh, etc)

### ISSUE THREAD

**Issue by desyncr** [desyncr/zpm-zsh#4]
- [ ] Support adding completions
- [ ] Support both binary and functions
- [ ] Support themes/prompts

**Comment by fennecdjay** [comment]
Correct me if I'm wrong, but aren't completion and binary/function support related to #3?

Also, I'm interrested in the themes/prompts support thing. I'd gladly help, if you provide information on the desired behavior.

**Comment by desyncr** [comment]
Yes, it's kinda related. #3 should be done first as zsh is our primary target.

A long term goal is to be shell agnostic.

DC*
Sent from mobile.
